indico/modules/categories/models/categories.py

```diff
@@ -486,6 +486,12 @@ def visible_categories_query(self):
         cte_query = Category.get_visible_categories_cte(self.id)
         return Category.query.join(cte_query, Category.id == cte_query.c.id)
 
+    def get_hidden_events(self, user=None):
+        """Get all hidden events within the given category and user."""
+        from indico.modules.events import Event
+        hidden_events = Event.query.with_parent(self).filter_by(visibility=0).all()
+        return [event for event in hidden_events if not event.can_display(user)]
```

Comment:
Isn't this the same or more explicit to write?
```suggestion
        return [event for event in hidden_events if not event.can_manage(user)]
```

Since `can_display` is `self.visibility != 0 or self.can_manage(user)` and we have `filter_by(visibility=0)`.

Reply:
Yes, but this is just because we do not have a specific permission at the moment. 
Consider a "Custom role", or "Security role" or "Focal point role" (that you do not have but we are working on them). All these roles will not have "event manage" permission, but a specific "can display" (and in the future: can('display', self, session.user).

Reply:
Sounds good then.

Reply:
Well... there was... ask Adrian :)

Reply:
Yeah I suggested removing it since the idea about adding `can(...)` is something for 3.1, i.e. a long time in the future ;)